src/couch_file.erl

```diff
@@ -524,6 +528,214 @@ handle_info({'EXIT', _, Reason}, Fd) ->
     {stop, Reason, Fd}.
 
 
+find_last_header(Fd, FileSize) ->
+    find_header_in_chunks(Fd, start_pos(FileSize), 1).
+
+start_pos(FileSize) when FileSize rem ?SIZE_BLOCK =:= 0 ->
+    FileSize;
+start_pos(FileSize) ->
+    %% set position to nearest block boundary past eof
+    %% if file size isn't multiple of block size
+    (FileSize div ?SIZE_BLOCK) * ?SIZE_BLOCK + ?SIZE_BLOCK.
+
+chunk_max_size() ->
+    config:get_integer("couchdb", "chunk_max_size", ?DEFAULT_CHUNK_MAX_SIZE).
```

Comment:
Seems sill to reconsult the config ets table multiple times for every iteration. Perhaps it would be cleaner to create a record for your current iteration state.